source/StartConditions.cpp

```diff
@@ -148,13 +144,49 @@ void StartConditions::Load(const DataNode &node)
 			else
 				child.PrintTrace("Skipping unrecognized attribute:");
 		}
+		else if(key == "on")
+		{
+			if(child.Token(1) == "display")
+				LoadState(child, StartState::VISIBLE);
+			else if(child.Token(1) == "reveal")
+				LoadState(child, StartState::REVEALED);
+			else if(child.Token(1) == "unlock")
+				LoadState(child, StartState::UNLOCKED);
+		}
 		else
 			conditions.Add(child);
 	}
-	if(description.empty())
-		description = "(No description provided.)";
-	if(name.empty())
-		name = "(Unnamed start)";
+	if(infoByState[StartState::UNLOCKED].description.empty())
+		infoByState[StartState::UNLOCKED].description = "(No description provided.)";
+	if(infoByState[StartState::UNLOCKED].name.empty())
+		infoByState[StartState::UNLOCKED].name = "(Unnamed start)";
+
+	system = GameData::Systems().Get(infoByState[StartState::UNLOCKED].system);
+	planet = GameData::Planets().Get(infoByState[StartState::UNLOCKED].planet);
+
+	// Fill missing information in states:
+	if(!infoByState[StartState::REVEALED].thumbnail)
+		infoByState[StartState::REVEALED].thumbnail = infoByState[StartState::UNLOCKED].thumbnail;
+	if(infoByState[StartState::REVEALED].name.empty())
+		infoByState[StartState::REVEALED].name = infoByState[StartState::UNLOCKED].name;
+	if(infoByState[StartState::REVEALED].description.empty())
+		infoByState[StartState::REVEALED].description = infoByState[StartState::UNLOCKED].description;
+	if(infoByState[StartState::REVEALED].system.empty())
+		infoByState[StartState::REVEALED].system = infoByState[StartState::UNLOCKED].system;
+	if(infoByState[StartState::REVEALED].planet.empty())
+		infoByState[StartState::REVEALED].planet = infoByState[StartState::UNLOCKED].planet;
+
+	if(!infoByState[StartState::VISIBLE].thumbnail)
+		infoByState[StartState::VISIBLE].thumbnail = infoByState[StartState::REVEALED].thumbnail;
+	if(infoByState[StartState::VISIBLE].name.empty())
+		infoByState[StartState::VISIBLE].name = infoByState[StartState::REVEALED].name;
+	if(infoByState[StartState::VISIBLE].description.empty())
+		infoByState[StartState::VISIBLE].description = infoByState[StartState::REVEALED].description;
+	if(infoByState[StartState::VISIBLE].system.empty())
+		infoByState[StartState::VISIBLE].system = infoByState[StartState::REVEALED].system;
+	if(infoByState[StartState::VISIBLE].planet.empty())
+		infoByState[StartState::VISIBLE].planet = infoByState[StartState::REVEALED].planet;
```

Comment:
Perhaps StartInfo or StartConditions should have some function which does this instead of needing to have the same stack of if statements twice.

Reply:
done, with a helper function